packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/queued_item_store/index_db/in_memory_queued_item_store.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:collection';
+
+import 'package:amplify_analytics_pinpoint_dart/src/impl/analytics_client/event_client/queued_item_store/queued_item_store.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.in_memory_queued_item_store}
+/// Stores string elements in device memory
+/// {@endtemplate}
+class InMemoryQueuedItemStore implements QueuedItemStore {
+  /// {@macro amplify_analytics_pinpoint_dart.in_memory_queued_item_store}
+  InMemoryQueuedItemStore();
+
+  int _lastId = 0;
```

Comment:
"lastId" indicates this is the ID of the last item in the DB (equal to _database.last.id), which it isn't.

```suggestion
  /// The next ID that should be used when adding an item in the DB.
  int _nextId = 0;
```